libraries/irule_diff.rb

```diff
@@ -0,0 +1,51 @@
+require 'set'
+
+# Diff 2 hashes to show the added, changed, and removed keys
+class IRuleDiff
+  class << self
+    MAX_IRULES = 100
+
+    def diff_hashes(target, current)
+      target_keys = Set.new target.keys
+      current_keys = Set.new current.keys
+      added = target_keys - current_keys
+      removed = current_keys - target_keys
+      changed = target.keys.select { |k| !current[k].nil? && target[k] != current[k] }
+      x = [added, changed, removed, target]
+      x
+    end
+
+    def diff(target, current_hash)
+      current_ordered = current_hash
+                          .to_a
+                          .sort{ |a, b| a[1] <=> b[1] }
+                          .map { |rule| rule.first }
```

Comment:
I think you're trying to get the keys of the hash sorted by their values? Just being a bit picky here since it took me a minute to reason through this, maybe there's a better way (It's Ruby, there's always a _different_ way at least)

`current_hash.sort_by { |_, val| val }.map(&:first)` might be more clear?